core/src/main/java/com/example/util/simpletimetracker/core/interactor/RecordTagViewDataInteractor.kt

```diff
@@ -59,13 +60,27 @@ class RecordTagViewDataInteractor @Inject constructor(
                 .takeIf { multipleChoiceAvailable }
                 ?.let(viewData::add)
 
-            available.map {
-                categoryViewDataMapper.mapRecordTag(
-                    tag = it,
-                    type = types[it.iconColorSource],
-                    isDarkTheme = isDarkTheme,
-                )
-            }.let(viewData::addAll)
+//            available.map {
+//                categoryViewDataMapper.mapRecordTag(
+//                    tag = it,
+//                    type = types[it.iconColorSource],
+//                    isDarkTheme = isDarkTheme,
+//                )
+//            }.let(viewData::addAll)
+
+            categoryViewDataMapper.groupToTagGroups(available).forEach { group ->
+                if (group.key.isNotEmpty()) {
+                    viewData.add(InfoViewData(text = group.key))
```

Comment:
this might be broken, I have no idea what a RecyclerView is 😆

Reply:
Should be ok. RecyclerView is a list of views that added programmatically, not by constructing whole list in the xml file.